sql/src/main/java/io/crate/operation/collect/files/FileReadingCollector.java

```diff
@@ -206,14 +206,15 @@ private void readLines(FileInput fileInput,
                            URI uri,
                            long startLine,
                            int retry) throws IOException {
-        InputStream inputStream = fileInput.getStream(uri);
-        if (inputStream == null) {
-            return;
-        }
-
-        String line;
         long linesRead = 0L;
-        try (BufferedReader reader = createReader(inputStream)) {
+        String line;
+
+        try (InputStream inputStream = fileInput.getStream(uri);
+             BufferedReader reader = createReader(inputStream)) {
```

Comment:
couldn't this cause createReader to fail if inputStream is null? Also I don't quite understand why you're closing inputStream instead of the BufferedReader. I don't really see how there was a resource leak here